### PR TITLE
Extension Exercise

### PR DESCRIPTION
**Pull request by NoyRobinson**
Extension exercise

### CODE REVIEW

File: src/main/kotlin/com/hibob/kotlinBasics/Extensions.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        result *= this
+    }
+
+    if(negative)
+        return 1 / result
+
+    return result
+}
+
+fun main(args: Array<String>) {
```
Comment (liorsiv973) [inline src/main/kotlin/com/hibob/kotlinBasics/Extensions.kt]: you can delete it in the pr it for your checks only

File: src/main/kotlin/com/hibob/kotlinBasics/Extensions.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+//       }
+//       return result
+//  }
+//
+//  fun main(args: Array<string>) {
+//      val sum = sum(listOf(1,2,3))
+//      print(sun) // 6
+//  }
+
+fun List<Int>.sum(): Int {
```
Comment (liorsiv973) [inline src/main/kotlin/com/hibob/kotlinBasics/Extensions.kt:19]: great! please check for future code the map{} function

File: src/main/kotlin/com/hibob/kotlinBasics/Extensions.kt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+
+// Exercise 2
+
+// Write the following program method as an infix extension function
+// fun toPowerOf(num: Number, exponent: Number): Double{
+//      return 0
+// }
+
+infix fun Double.toPowerOf(exponent: Int): Double {
```
Comment (liorsiv973) [inline src/main/kotlin/com/hibob/kotlinBasics/Extensions.kt]: maybe do the extension on Number so all types of numbers can use it. Also there is a much easier solution if you will use the Math library

**Comment by NoyRobinson** [comment]
Thanks!
I corrected my code following your comments

**Review by liorsiv973 — APPROVED** [review]

**Pull request merged**